# DESAFIO -11
## ‚úÖ Descri√ß√£o  
## ultiza√ß√£o da ferramenta Trivy para fazer uma verifica√ß√£o de vunerabilidade em uma imagem python:3.9 üî¨
## - 1 ao executar o comando `docker run --rm -v //var/run/docker.sock:/var/run/docker.sock aquasec/trivy:latest image python:3.9-alpine` temos a seguinte saida :
![alt text](images/image.png) 
![alt text](images/image2.png)
## - 2 a saida nos mostra a biblioteca, SQL-Lite, que est√° com uma vunerabilidade e a vers√£o que tem esse erro corrigido :
 ![alt text](images/image3.png) 
 # ‚ùåvers√£o com erro 3.48.0-r0
 ## O produto realiza um c√°lculo que pode causar um estouro de inteiro, assumindo incorretamente que o resultado ser√° sempre maior que o valor original. Isso pode afetar o controle de recursos ou a execu√ß√£o do sistema.
 # ‚úÖ vers√£o corrigida 3.48.0-r1
 ## Para corrigir esse erro, pode usar o comando abaixo no seu Dockerfile.
 ```docker
 RUN apk update && apk upgrade --no-cache
 ```
## Ou para garantir que est√° instalando a vers√£o espec√≠fica e corrigida do SQLite.
 ```docker
RUN apk add --no-cache sqlite-libs=3.48.0-r1
 ```
## -3 No segundo bloco temos as vunerabilidades das bibliotecas `pip` e `setuptools` 
![alt text](images/image4.png)
# ‚ùåvers√£o com erro pip 23.0.1
## Em vers√µes do pip anteriores √† 23.3, ao instalar pacotes via Mercurial (hg+...), era poss√≠vel injetar configura√ß√µes maliciosas no comando hg clone, podendo alterar o reposit√≥rio instalado. Essa falha ocorre por n√£o sanitizar corretamente as entradas externas usadas na constru√ß√£o do comando. Usu√°rios que n√£o utilizam o Mercurial n√£o s√£o afetados.
# ‚úÖ vers√£o corrigida pip 23.3
## para corrigir basta adicionar no dockerfile o seguinte comando :
 ```docker
 RUN pip install --no-cache-dir pip==23.3
 ```
# ‚ùåvers√£o com erro setuptools 58.1.0
## ReDoS (corrigido na 65.5.1):
## A vers√£o vulner√°vel do setuptools permite a execu√ß√£o remota de c√≥digo por meio de fun√ß√µes de download mal protegidas, acesso indevido a arquivos do sistema por meio da manipula√ß√£o de caminhos no PackageIndex e travamento do sistema causado pelo uso de uma express√£o regular ineficiente no arquivo package_index.py, que pode ser explorada com entradas maliciosas.
# ‚úÖ vers√£o corrigida setuptools 78.1.1
## para corrigir basta adicionar no dockerfile o seguinte comando :
 ```docker
RUN pip install --no-cache-dir setuptools==78.1.1
 ```